extensions/github-repository-search/src/index.tsx

```diff
@@ -11,55 +11,113 @@ import {
+    });
+  }
+  if (fetchUserDataError) {
+    showToast({
+      style: Toast.Style.Failure,
+      title: "Failed fetching user and organizations",
+      message: fetchUserDataError instanceof Error ? fetchUserDataError.message : String(fetchUserDataError),
     });
   }
 
   const isLoading = searchText !== debouncedSearchText || isLoadingVisitedRepositories || isLoadingRepositories;
 
   return (
-    <List isLoading={isLoading} onSearchTextChange={setSearchText}>
+    <List
+      isLoading={isLoading}
+      onSearchTextChange={setSearchText}
+      searchBarAccessory={<FilterDropdown userData={userData} onFilterChange={setSearchFilter} />}
+    >
       <List.Section
         title="Visited Repositories"
         subtitle={visitedRepositories ? String(visitedRepositories.length) : undefined}
       >
         {visitedRepositories
-          ?.filter((r) => r.nameWithOwner.includes(searchText ?? ""))
+          ?.filter(
+            (r) =>
+              r.nameWithOwner.includes(searchText ?? "") &&
+              r.nameWithOwner.match(
+                // Converting query filter string to regexp:
+                `${searchFilter?.replaceAll(/org:|user:/g, "").replaceAll(" ", "|")}/.*`
+              )
+          )
           .map((repository) => (
             <RepositoryListItem key={repository.id} repository={repository} onVisit={visitRepository} />
           ))}
       </List.Section>
-      <List.Section title="Found Repositories" subtitle={data ? String(data.repositoryCount) : undefined}>
-        {data?.nodes?.map((repository) => (
+      <List.Section
+        title="Found Repositories"
+        subtitle={repositories ? String(repositories.repositoryCount) : undefined}
+      >
+        {repositories?.nodes?.map((repository) => (
           <RepositoryListItem key={repository.id} repository={repository} onVisit={visitRepository} />
         ))}
       </List.Section>
     </List>
   );
 }
 
+function FilterDropdown({
+  userData,
+  onFilterChange,
+}: {
+  userData: UserDataResponse["viewer"] | undefined;
+  onFilterChange: (filter: string) => void;
+}) {
+  return (
+    <List.Dropdown tooltip="Filter Repositories" storeValue={true} onChange={onFilterChange}>
+      <List.Dropdown.Item title={"All Repositories"} value={""} />
+      {userData && (
+        <List.Dropdown.Item
+          title={"My Repositories"}
+          value={`user:${userData.login} ${userData.organizations.nodes.map((org) => `org:${org.login}`).join(" ")}`}
+        />
+      )}
+      {userData && (
+        <List.Dropdown.Section>
+          <List.Dropdown.Item title={userData.login} value={`user:${userData.login}`} />
+          {userData.organizations.nodes.map((org) => (
+            <List.Dropdown.Item key={org.login} title={org.login} value={`org:${org.login}`} />
+          ))}
+        </List.Dropdown.Section>
+      )}
+    </List.Dropdown>
+  );
+}
```

Comment:
How about not returning a dropdown if we don't have any user data? A dropdown with just a single item doesn't make much sense.
```suggestion
  return userData ? (
    <List.Dropdown tooltip="Filter Repositories" storeValue={true} onChange={onFilterChange}>
      <List.Dropdown.Item title="All Repositories" value="" />
        <List.Dropdown.Item
          title="My Repositories"
          value={`user:${userData.login} ${userData.organizations.nodes.map((org) => `org:${org.login}`).join(" ")}`}
        />
        <List.Dropdown.Section>
          <List.Dropdown.Item title={userData.login} value={`user:${userData.login}`} />
          {userData.organizations.nodes.map((org) => (
            <List.Dropdown.Item key={org.login} title={org.login} value={`org:${org.login}`} />
          ))}
        </List.Dropdown.Section>
    </List.Dropdown> : null
  );
}
```